submissions/Test123423235/CarRacingGame/Renderer.java

```diff
@@ -65,19 +65,19 @@ public boolean addRenderObject(IRenderObject attachment){
         }
 
         int stored = -1;
-        for ( int i = 0; i < Store.length; i ++){
```

Comment:
To add an object only if it is not already in the array you could use something like this:

```
public boolean addRenderObject(IRenderObject o) {
        int potentialSpace = -1;
        for(int i = 0; i < objects.length; i++) {
            if (objects[i] == null) {
                potentialSpace = i;
            }
            if(objects[i] != null && objects[i].equals(o)) {
                return false;
            }
        }
        if (potentialSpace != -1) {
            objects[potentialSpace] = o;
        } else {
// You can replace this part with your own expand function
            IRenderObject[] tmp = new IRenderObject[objects.length + 1]; 
            System.arraycopy(objects, 0, tmp, 0, objects.length);
            objects = tmp;
            objects[objects.length - 1] = o;
        }
        return true;
    }

```